astropy/units/tests/test_format.py

```diff
@@ -48,8 +48,8 @@ def test_unit_grammar(strings, unit):
     "string", ["sin( /pixel /s)", "mag(mag)", "dB(dB(mW))", "dex()"]
 )
 def test_unit_grammar_fail(string):
+    print(string)
     with pytest.raises(ValueError):
-        print(string)
```

Comment:
I'm not entirely sure, I can remove it.

Reply:
I think calling the `__str__()` or `__repr__()` methods should trigger the desired error without having to call the entire `print()` function.

Reply:
Alright ill try out the __repr__() method on these tests.

Reply:
Yes, for some tests like the one above it would print just the enumeration of parameters like `test_unit_grammar[strings9-unit9]`, so the actual string would add a bit of convenience, but for single parametrisation even that should not be necessary.